src/KafkaFlow.OpenTelemetry/OpenTelemetryProducerEventsHandler.cs

```diff
@@ -0,0 +1,110 @@
+﻿namespace KafkaFlow.OpenTelemetry
+{
+    using System;
+    using System.Diagnostics;
+    using System.Linq;
+    using System.Text;
+    using System.Threading.Tasks;
+    using global::OpenTelemetry;
+    using global::OpenTelemetry.Context.Propagation;
+
+    internal static class OpenTelemetryProducerEventsHandler
+    {
+        private const string PublishString = "publish";
+        private static readonly TextMapPropagator Propagator = Propagators.DefaultTextMapPropagator;
+
+        public static Task OnProducerStarted(IMessageContext context)
+        {
+            try
+            {
+                var activityName = !string.IsNullOrEmpty(context?.ProducerContext.Topic) ? $"{context?.ProducerContext.Topic} {PublishString}" : PublishString;
+
+                // Start an activity with a name following the semantic convention of the OpenTelemetry messaging specification.
+                // The convention also defines a set of attributes (in .NET they are mapped as `tags`) to be populated in the activity.
+                // https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/messaging.md
+                var activity = KafkaFlowActivitySourceHelper.ActivitySource.StartActivity(activityName, ActivityKind.Producer);
```

Comment:
```suggestion
var activity = KafkaFlowActivitySourceHelper.ActivitySource.StartActivity(activityName, ActivityKind.Producer) ?? Activity.Current;

 // Since the next commands depend on data from the activity, if the activity is null, should we skip the rest of the execution? 
if(activity = null)
{
    return;
}
```

Reply:
Activity is only supposed to be created if there are any listeners available
https://github.com/dotnet/runtime/blob/main/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/ActivitySource.cs#L137